5 - RPG/Code/ClassLibrary1/CharacterBuilder.cs

```diff
@@ -0,0 +1,58 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace Characters
+{
+    public class CharacterBuilder
+    {
+
+
+            public Character BuildCharacter(int head, FitnessLevel fit, int heightInInches, List<AbilityPoints> distributedPoints)
+            {
+
+            int expectedAbilityPointTotal = 60;
+
+                if (head < 0 || head > 100)
+                {
+                    throw new Exception("Unable to create character headShape should be between 0 and 100.\n");
+                }
+
+                Dictionary<string, AbilityPoints> validatedAbilities = new Dictionary<string, AbilityPoints>();
+
+                string errormessages = "";
+                var total = 0;
+                foreach (AbilityPoints ability in distributedPoints)
```

Comment:
If you pulled out the `total` part below and set up the ability objects to have a method `IsValid` that returns a bool you could simplify your method to be something like:
```
if (distributedPoints.All(x => x.IsValid)
{
  // Build Character
}
else
{
  var errorMessages = distributedPoints.Select(x => x.GetValidationErrorMessage());
  var error = errorMessages.Join("\n");
  throw new Exception(error)
}
```